src/polyglot_ffi/parsers/ocaml.py

```
@@ -273,8 +273,10 @@ def _parse_function(
             func = self._parse_signature(full_sig.strip(), start_line)
             return func, doc, lines_consumed
         except ParseError as e:
-            # Re-raise with line info
-            raise ParseError(str(e), start_line)
+            # Re-raise with line info (avoid duplicating if already has line info)
+            if e.context.line:
+                raise  # Already has line info, just re-raise
+            raise ParseError(e.message, line=start_line)
```

Ruff (0.14.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)